eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -113,11 +113,16 @@ $onboardedPackages = &$GetOnboardedDocsMsPackagesFn `
 # because we need to generate ToCs for packages which are not necessarily "New"
 # in the metadata AND onboard legacy packages (which `Update-DocsMsPackages.ps1`
 # does not do)
-$metadata = (Get-CSVMetadata).Where({
-    $_.Package `
-      -and $onboardedPackages.ContainsKey($_.Package) `
-      -and $_.Hide -ne 'true'
-  })
+$fullMetadata = Get-CSVMetadata
+$metadata = @()
+foreach($metadataEntry in $fullMetadata) {
```

Comment:
Should we instead loop through the onboardedPackages and find the matching metadata instead?

Reply:
We can build a package look via GetPackageLookup like we are doing below.

Reply:
`onboardedPackages` is a lookup map, and metadata is csv array. So this is an efficiency comparison. 
The script is used by python, js, java already. I'd prefer to make few changes to avoid mess up the work in progress.

Reply:
OK Yes I guess we are just trying to find the intersection of the 2 sets so either direction works. I will leave this review up to @danieljurek the change but I if feels like we are reading the metadata from the csv multiple times in this function. Just potentially an optimization to be had.